client/lib/feed-stream-store/feed-stream.js

```diff
@@ -188,6 +189,13 @@ export default class FeedStream {
 			this.selectedIndex = nextIndex;
 			this.emitChange();
 		}
+
+		// If we are getting close to the end of the loaded stream, or are already at the end,
+		// start fetching new posts
+		if ( nextIndex + 4 > this.postKeys.length || nextIndex === -1 ) {
```

Comment:
Magic number :)

Can we define that separately and add a comment to explain why it's 4?